pkg/gardener/terraformer/terraformer.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package terraformer
+
+import (
+	gardenerterraformer "github.com/gardener/gardener/pkg/operation/terraformer"
+	"github.com/sirupsen/logrus"
+	"k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type terraformer struct {
```

Comment:
This type and its methods are not needed. `*gardenerterraformer.Terraformer` already implements `Interface`

Reply:
Not exactly: `Interface` expects the terraformer `Interface` as return type for some functions. Go won't do automatic casts if your implementation returns the concrete type, thus the interface is required.

Reply:
How about changing the `Interface` type so that `*gardenerterraformer.Terraformer` conforms? Go would do automatic casts on the concrete types returned by the methods in this case.

Reply:
I don't think you can change the interface that way: Take the `InitializeWith` method for example: In Gardener, this is `InitializeWith(initializer Initializer) *Terraformer`. It always returns the concrete `*Terraformer` type. Since we want to be able to return arbitrary mock implementations with the `Interface`, our definition for the `InitializeWith` method has to return the `Interface`, otherwise it's useless. Also I see that the methods might be the same, but as said: go does *not* do automatic casts, there simply isn't a thing like that. If you see a way of doing this let me know.

Reply:
That's right, I didn't think about it. OK, please ignore these comments then.

Reply:
I'll think about it when I have time, but that might take a few days, I am at KubeCon this week.